pageserver/src/tenant/layer_map.rs

```diff
@@ -276,17 +277,17 @@ where
     pub(self) fn insert_historic_noflush(&mut self, layer: Arc<L>) -> anyhow::Result<()> {
         let key = historic_layer_coverage::LayerKey::from(&*layer);
         if self.historic.contains(&key) {
-            bail!(
+            error!(
                 "Attempt to insert duplicate layer {} in layer map",
                 layer.short_id()
             );
-        }
-        self.historic.insert(key, Arc::clone(&layer));
+        } else {
+            self.historic.insert(key, Arc::clone(&layer));
```

Comment:
Just noticed that this (and previous PR) bypasses the buffer, so I wonder if there are some reconstruction steps being missed when the BufferedUpdates commit on drop?